.github/workflows/stackhpc-container-image-build.yml

```diff
@@ -207,6 +215,109 @@ jobs:
           path: ${{ matrix.distro }}-container-images
           retention-days: 7
 
+      - name: Generate list of images to scan/push
+        run: |
+          # Clean up any stale data
+          rm -rf ${{ matrix.distro }}-docker-images.txt
+
+          # Make a file of imagename:tag
+          grep --invert-match --no-filename ^REPOSITORY ${{ matrix.distro }}-container-images |\
+          sed 's/ \+/:/g' |\
+          cut -f 1,2 -d: > ${{ matrix.distro }}-docker-images.txt
+        
+      - name: Scan built container images
+        run: |
+          set -euo pipefail
+
+          # Clean any stale data
+          rm -rf ${{ matrix.distro }}-image-scan-output
+          rm -f images-to-push.txt
+
+          # Make a fresh output directory
+          mkdir -p ${{ matrix.distro }}-image-scan-output
+
+          # If Trivy detects no vulnerabilities, add the image name to images-to-push.txt.
+          # If there are vulnerabilities detected, generate a CSV summary and do not add to
+          # images-to-push.txt.
+          while read -r image; do
+            filename=$(basename $image | sed 's/:/\./g')
+            if $(trivy image \
```

Comment:
I tried a build of glance-api to test this out. It passed on Ubuntu but failed on CentOS Stream 8:

```
2023-09-27T11:46:49.285Z	FATAL	image scan error: scan error: unable to initialize a scanner: unable to initialize a docker scanner: 4 errors occurred:
	* unable to inspect the image (ark.stackhpc.com/stackhpc-dev/centos-source-bifrost-deploy:yoga-20230927T111042): permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/ark.stackhpc.com/stackhpc-dev/centos-source-bifrost-deploy:yoga-20230927T111042/json": dial unix /var/run/docker.sock: connect: permission denied
	* failed to initialize a containerd client: failed to dial "/run/containerd/containerd.sock": connection error: desc = "transport: error while dialing: dial unix /run/containerd/containerd.sock: connect: permission denied"
	* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory
	* GET https://ark.stackhpc.com/v2/stackhpc-dev/centos-source-bifrost-deploy/manifests/yoga-20230927T111042: UNAUTHORIZED: Insufficient permissions; map[]


jq: error: Could not open file centos-image-scan-output/centos-source-bifrost-deploy.yoga-20230927T111042.json: No such file or directory
```

I suppose cloud-user isn't in the docker group, so can't access the docker socket.

It also shows that we're not distinguishing between a failure to scan vs. a match on high/critical vulnerabilities.

Reply:
https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/6325114889/job/17175971886

Reply:
The runners have changed so much, I don't think this is a relevant issue anymore.